util/http.go

```diff
@@ -182,8 +186,11 @@ func MarshalResponse(r *http.Request, value interface{}, allowed []EncodingType)
 	if protoIdx < jsonIdx && protoIdx < yamlIdx {
 		// Protobuf-encode the config.
 		contentType = ProtoContentType
-		if body, err = gogoproto.Marshal(value.(gogoproto.Message)); err != nil {
-			err = Errorf("unable to marshal %+v to protobuf: %s", value, err)
+		msg, ok := value.(gogoproto.Message)
+		if !ok {
+			err = Errorf("unable to convert %+v to protobuf", value)
```

Comment:
We shouldn't return an error if someone sends `Accept: protobuf` when we can't support it (we don't reject requests that have completely unknown values in the accept header). Move the gogoproto.Message conversion earlier and if it fails set protoIdx to MaxInt32 (just as if protobuf weren't in the accept header at all).